core/server/master/src/main/java/alluxio/master/block/meta/MasterWorkerInfo.java

```diff
@@ -278,6 +278,9 @@ public WorkerInfo generateWorkerInfo(Set<WorkerInfoField> fieldRange, boolean is
         case ADDRESS:
           info.setAddress(mMeta.mWorkerAddress);
           break;
+        case BLOCK_COUNT:
```

Comment:
@maobaolong @yuzhu A shared lock on `MasterWorkerInfo.mBlockListLock` is missing here. You can find how this method should be properly locked in the method comment.

Reply:
sure

Reply:
@jiacheliu3 Thanks for your remind, would you lease explain more why lost `MasterWorkerInfo.mBlockListLock` here, and what will happen if missing `mBlockListLock `

Reply:
@maobaolong You can see the comments in `MasterWorkerInfo` where the locking mechanism of this class is further explained. If you miss the read lock when reading the block list you may see inconsistent result.

Reply:
@jiacheliu3 Since there I just read the block count here, I know `mBlocks` is guarded by `mBlockListLock`, and `mBlocks.size()` is more heavy if guarded by `mBlockListLock`, and we get the size just want to expose it for metrics  purpose, so, can we expose a inconsistent block count by this not thread-safe method?